handler/helpers/accounts/accounts.go

```diff
@@ -56,16 +59,18 @@ type Svc struct {
 	iamSvc           iamiface.IAMAPI
 	organizationsSvc organizationsiface.OrganizationsAPI
 	downloaderSvc    s3manageriface.DownloaderAPI
+	stsSvc           stsiface.STSAPI
 }
 
 // NewAccountsSvc ... creates new Svc struct
 func NewAccountsSvc(cfg client.ConfigProvider) (as *Svc, err error) {
-	if cfg == nil {
+	if cfg == nil || reflect.ValueOf(cfg).IsNil() {
```

Comment:
Is there a situation where cfg would not be nil, but reflect.ValueOf(cfg) would be?

Reply:
That is an interesting question.  When I was writing the test I discovered cfg will never be nil because client.ConfigProvider is an interface{} type it will have a type and a value.

Reply:
cfg has a value of nil, but cfg == nil is false because of its type?

Reply:
This is related to the default value associated with `sess` in his example code above. If he called `NewAccountsSvc(nil)` explicitly with `nil` we would follow the `cfg == nil` path. However, the default value of `var sess session.Session` is not really `nil`, as it does have an address assigned in memory.

Reply:
I think we should just use the interface. The caller should use the interface and a factory method and we should do the same. The only other logical option I see is to follow a null object pattern and attach an isNil method to cfg. 

I think we can agree that two checks isn't something we want to propagate if we can avoid it - especially since it complicates testing. In this case did we test all branches of line 67? I think we'd actually have to add another test case that uses the interface to do so.

Reply:
We had a conversation about this and have decided it would be better to panic than obscure these situations by 'gracefully' handling `nil`.

Reply:
Since passing a nil is impossible the way we are consuming NewAccounsSvc(), we are moving the nil check.